include/boost/math/interpolators/cubic_b_spline.hpp

```diff
@@ -0,0 +1,67 @@
+// Copyright Nick Thompson, 2017
+// Use, modification and distribution are subject to the
+// Boost Software License, Version 1.0.
+// (See accompanying file LICENSE_1_0.txt
+// or copy at http://www.boost.org/LICENSE_1_0.txt)
+
+// This implements the compactly supported cubic b spline algorithm described in
+// "Numerical Analsis, Graduate Texts in Mathematics 181", by Rainer Kress, section 8.3.
```

Comment:
Pedantry is alive and well at Boost.Math ;-).  I'll make sure that the docs will receive some editorial work in due course to fill out full  references, and links to sources on the web where available, and links to examples, example code etc.  So I hope that John will get this on the develop branch, or a develop/interpolation branch as asoon as convenient.